### PR TITLE
OCPBUGS-19073: Move KAS access label to right place

### PR DESCRIPTION
**Pull request by gnufied**
Fixes https://issues.redhat.com/browse/OCPBUGS-19073

### CODE REVIEW

File: assets/csidriveroperators/aws-ebs/hypershift/mgmt/deployment.patch.yaml [hunk 1]
```diff
@@ -2,10 +2,11 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: aws-ebs-csi-driver-operator
-  labels:
-    hypershift.openshift.io/need-management-kas-access: true
 spec:
   template:
+    metadata:
+      labels:
+        hypershift.openshift.io/need-management-kas-access: true
```
Comment (jsafrane) [inline assets/csidriveroperators/aws-ebs/hypershift/mgmt/deployment.patch.yaml:9]: missing quotes around `true`.

> E1002 22:02:46.995956       1 runtime.go:79] Observed a panic: &json.UnmarshalTypeError{Value:"bool", Type:(*reflect.rtype)(0x254b2c0), Offset:437, Struct:"ObjectMeta", Field:"spec.template.metadata.labels"} (json: cannot unmarshal bool into Go struct field ObjectMeta.spec.template.metadata.labels of type string)
goroutine 837 [running]:

Comment (gnufied) [reply assets/csidriveroperators/aws-ebs/hypershift/mgmt/deployment.patch.yaml]: fixed.

**Comment by gnufied** [comment]
Now it is failing on:


```
    util.go:807: 
        Unexpected error:
            <*errors.errorString | 0xc0006e6000>: 
            NeedManagementKASAccessLabel label is not allowed on component: aws-ebs-csi-driver-operator
            {
                s: "NeedManagementKASAccessLabel label is not allowed on component: aws-ebs-csi-driver-operator",
            }
        occurred
            --- FAIL: TestCreateClusterPrivate/EnsureHostedCluster/EnsureNetworkPolicies/EnsureComponentsHaveNeedManagementKASAccessLabel (0.01s)
```

I am checking with hypershift team about this.

**Pull request merged**